### ISSUE TITLE
JavaCodeCompletion module fails to compile because of API change

### ISSUE THREAD

**Issue by kbowers-ibm** [apache/incubator-kie-issues#114]
It seems that an API change on JDT.LS project is incompatible with our codebase.

<img width="1118" alt="Screenshot 2023-05-11 at 12 49 09" src="https://github.com/kiegroup/kie-issues/assets/16005046/b48f8062-96fb-4a49-88af-4458e63d3bcf">

To solve that, I'm going to split this into multiple sub-tasks:
- Using the tag version 1.23.0 instead of the snapshot one. That will solve the issue. 
- Unfortunately, tagged versions get stale after some months (see https://github.com/eclipse/eclipse.jdt.ls/issues/1970) So, we need to update our codebase accordingly with the API changes

```[tasklist]
### Acceptance criteria
- [ ] https://github.com/kiegroup/kie-tools/pull/1626
- [ ] https://github.com/kiegroup/kie-tools/pull/1663
- [ ] https://github.com/kiegroup/kie-tools/pull/1688
- [ ] Update our codebase accordingly with the API changes (Using TypeHierarchy message?)
```

**Comment by kbowers-ibm** [comment]
Duplicate

**Comment by tiagobento** [comment]
Waiting on https://github.com/eclipse/eclipse.jdt.ls/pull/2033
